### ISSUE TITLE
Navigation doesn’t start at any point when meeting existing track

### ISSUE THREAD

**Issue by C6competition** [osmandapp/OsmAnd#14796]
Assuming you have a track of about 200 km and your position is somewhere in between start and finish( I tested about half way) and near or on the track. Now start the navigation (navigation is set to car mode / no new calculation) for this track. I expect the navigation will lock onto the track, as soon as I am on the track. But nothing happens. I get no pictograms in the left panel nor announcements to follow the track. T

The track is all the time to see and I could follow it, but never get linked to it.  It works of course fine, when you meet the start point. Is there any reason, this should not work?

**Comment by C6competition** [comment]
Assuming you have a track of about 200 km and your position is somewhere in between start and finish( I tested about half way) and near or on the track. Now start the navigation (navigation is set to car mode / no new calculation) for this track. I expect the navigation will lock onto the track, as soon as I am on the track. But nothing happens. I get no pictograms in the left panel nor announcements to follow the track. The track is all the time to see and I could follow it, but never get linked to it.  It works of course fine, when you meet the start point. Is there any reason, this should not work? 

Best regards

**Comment by C6competition** [comment]
Sorry, Idon't wanted to close issue

**Comment by C6competition** [comment]
After some more testing, I noticed, if the navigation mode is switched to **recalculate**, then it does, what I expect it to do. The navigation leads me to the nearest or start point of the track, corresponding to what You choose at navigation/option/follow track. 
If navigation mode **not recalculate** is selected, the navigation always begins at the start of the track, no matter, what You choose at navigation/option/follow track.
Up to me **not recalculate**  means, the track itself will not be changed, when I am leaving the track during active navigation. This doesn’t mean, when  starting the navigation, I don’t get locked onto the track, if I am already on the track or don’t get lead to the nearest point of the track, if I am away from the track.... Otherwise the option under navigation/option/follow track  would make no sense as well.
To me this looks like a bug, it would be very helpful to get it fixed.

**Comment by Zirochkabila** [comment]
Your case was tested with application version 4.3.0#13642mqta, released: 2022-07-11 - it could not be reproduced.
On a route of about 200 km, when you continue driving from the middle of the route, the navigation is fixed on the track.

**Comment by C6competition** [comment]
I looked for updates because the version I used 4.1.11. The latest version I found is 4.2.6! So I used this one to make the following example:
[Route 1_2 Goldbeck 205 Km.ZIP](https://github.com/osmandapp/OsmAnd/files/9084003/Route.1_2.Goldbeck.205.Km.ZIP)

Route & Position on map

![1 Route   Position on Map](https://user-images.githubusercontent.com/99248549/178265369-b98420ca-3c81-4665-b263-1975157a4630.jpg)

Route Select

![2 Route Select](https://user-images.githubusercontent.com/99248549/178265449-7e7ed862-931c-4001-863b-bd9b05c57c0b.jpg)

Select Options

![3 Select Setting](https://user-images.githubusercontent.com/99248549/178265521-54a79615-5ef5-4d54-be3c-6ef38f6097a4.jpg)

Navigation Setting

![4 Navigation Setting](https://user-images.githubusercontent.com/99248549/178265617-fee4f90b-bc17-45a2-9487-8dde3a76929c.jpg)

Select No Recalculation

![5 Select No Recalculation](https://user-images.githubusercontent.com/99248549/178265706-04cfc389-1f52-40e5-8aef-b3c13168279a.jpg)

Select Meeting Point

![6 Select Meeting Point](https://user-images.githubusercontent.com/99248549/178265779-02c3da6c-7386-49f4-9c0c-16f737c8e17f.jpg)

Start Navigation

![7 Start Navigation No Recalc](https://user-images.githubusercontent.com/99248549/178265840-d4062e68-be08-4767-8325-13fa21661c78.jpg)

Result without Recalculation

![8 No Join After Start](https://user-images.githubusercontent.com/99248549/178265929-b166d858-d575-4e64-a2a1-313865b6f252.jpg)

Change Navigation Parameter

![9 Change Route Parameter](https://user-images.githubusercontent.com/99248549/178266041-0b23fbe9-2a3c-4106-8331-b16ba82252ed.jpg)

Start Navigation with Recalc

![10 Start Navigation Recalc](https://user-images.githubusercontent.com/99248549/178266249-b0e9ae64-11f2-499e-96a4-92cd74cbc5f9.jpg)

Result with Reclaculation

![11 Join Route After Start](https://user-images.githubusercontent.com/99248549/178266312-19434c77-cb18-4806-b462-24ebbee7a337.jpg)

As You can see, there is a difference.

By updating my sytem, I noticed, if you want to select the route via the 'Follow Track' menu, before updating you could select the track out of all track folders and even more helpfull: visible tracks. The last option is very usefull because now you have to scroll down all routes! Will this be back in newer versions?

Best regards

**Comment by Zirochkabila** [comment]
Having tested with version 4.3.0#13644mqta, released: 2022-07013 - no such situation was found. The route is built from any point regardless of whether Recalculate route is activated or deactivated.

Try resetting the profile to default and check again.
Also, try to use another GPX.

Information about Follow track menu has been forwarded to our developers for review, thanks.

**Comment by vshcherb** [comment]
> Up to me not recalculate means, the track itself will not be changed, when I am leaving the track during active navigation. This doesn’t mean, when starting the navigation, I don’t get locked onto the track, if I am already on the track or don’t get lead to the nearest point of the track, if I am away from the track

Not recalculate actually related only to the navigation and indeed it could be confusing when you enable navigation by GPX. We assume that if you enable navigation by GPX - track never will be recalculated and you will be always advised to reach the closest point of the track.

**Comment by C6competition** [comment]
Okay, thank You @vshcherb, so my assumption was correct.

I will go on investigating, why this doesn't work in my case.
To have the same version You use, could You please send me 4.3.0,
because in the internet I only found 4.2.6

Best regards

**Comment by C6competition** [comment]
Hallo @Zirochkabila ,

I followed your advise, reset the profile (motorcycle) switched to no recalculation and loaded a different GPX.
The result is the same as before!

New GPX is this one:
[Route 2 Barntrup 260 Km.zip](https://github.com/osmandapp/OsmAnd/files/9110962/Route.2.Barntrup.260.Km.zip)

If I use a short route (only about 30 km long) everything works fine!



So I made a new route following exactly "Route 2 Barntrup 260 Km" (draw new on top of it) and shortened it step by step
until it works.

The generated route are the following:
[Route 2 Barntrup shortend 1.zip](https://github.com/osmandapp/OsmAnd/files/9110974/Route.2.Barntrup.shortend.1.zip)
[Route 2 Barntrup shortend 2.zip](https://github.com/osmandapp/OsmAnd/files/9110975/Route.2.Barntrup.shortend.2.zip)
[Route 2 Barntrup shortend 3.zip](https://github.com/osmandapp/OsmAnd/files/9110976/Route.2.Barntrup.shortend.3.zip)

Shortened route 1 & 2 still have the fault, while route 3 is working correctly!
As you see, route 3 is only a few km longer!

This is my current position:

![screenshot 2](https://user-images.githubusercontent.com/99248549/178964504-d6821024-3c22-48d4-97e0-fa9f0581e354.jpg)

Do You have another suggestion?

**Comment by C6competition** [comment]
Hallo @vshcherb 


Only to make sure, we are talking about the same thing:

Both excamples I use the same GPX-Track, which is continuosly on the street "Gerader Weg".
In every example I leave the "Gerader Weg" , drive into "Spembrink" via "Eikser Grund" via "Kapellenweg" back to the  GPX-Track on the "Gerader Weg".


1.	Recalculation
Assume You are navigating and following a GPX –Track. If for any reason you leave the GPX-Track,
the navigation starts to calculate a new route, to lead you back to the GPX-Track!


Look here:

![Screenshot_20220715-144532](https://user-images.githubusercontent.com/99248549/179234658-e36e72dd-7493-43ef-b3a9-6610d44ddef1.png)
![Screenshot_20220715-145845](https://user-images.githubusercontent.com/99248549/179234667-d886877e-7e50-42ce-8476-2019cb2d7384.png)
![Screenshot_20220715-145922](https://user-images.githubusercontent.com/99248549/179234669-0e437cb7-1c67-4434-95a7-1e3a6c0b173f.png)
![Screenshot_20220715-150104](https://user-images.githubusercontent.com/99248549/179234675-1f754570-272c-414c-a525-5f126dbdb688.png)
![Screenshot_20220715-150122](https://user-images.githubusercontent.com/99248549/179234677-84ee0b81-1b66-4a6c-9d08-13d1b4240a62.png)
![Screenshot_20220715-150210](https://user-images.githubusercontent.com/99248549/179234680-60baa453-9ad5-4600-8b77-fd079a105d75.png)
![Screenshot_20220715-150228](https://user-images.githubusercontent.com/99248549/179234685-2c2085a4-d104-4297-bdf6-68e8ad879337.png)


2.	 No Recalculation

Assume You are navigating and following a GPX –Track. If for any reason you leave the GPX-Track,
you get the hint, you left the GPX-Track! But there is no route calculated, which leads you back to the GPX-Track.


Look here:

![Screenshot_20220715-150553](https://user-images.githubusercontent.com/99248549/179235411-e874f447-38f2-4580-b29e-344c3900ed18.png)
![Screenshot_20220715-150638](https://user-images.githubusercontent.com/99248549/179235435-b321d3be-dabd-468a-b5f1-ee5ae532fce1.png)
![Screenshot_20220715-150702](https://user-images.githubusercontent.com/99248549/179235458-423a8a4b-c247-4700-a69a-ce194b8f877e.png)
![Screenshot_20220715-150723](https://user-images.githubusercontent.com/99248549/179235482-8c2bdaa8-710a-4c3f-9307-1b89e755881d.png)
![Screenshot_20220715-150749](https://user-images.githubusercontent.com/99248549/179235500-81ff26ed-8857-46b8-b59f-a22a285d1698.png)
![Screenshot_20220715-150819](https://user-images.githubusercontent.com/99248549/179235518-0a34af45-f153-4dfc-b323-7a0c015f6963.png)
![Screenshot_20220715-150844](https://user-images.githubusercontent.com/99248549/179235538-7a2bf24a-c624-42c9-94c0-78c267264a98.png)

This is to my understanding the navigation difference between Recalculation and Non Recalculation!
There is no relationship on how or where to enter a GPX-Track at start of navigation!

Do You confirm?

**Comment by C6competition** [comment]
In the meantime I installed the new Osmand+ version 4.2.7 on a other phone Samsung A520F with Android 8.0.0 and tested the issue again:

My position for life and simulate tests:
![screenshot 3](https://user-images.githubusercontent.com/99248549/180642486-9c51c03d-f0ad-43db-979d-1fd80b00371b.jpg)

My tracks for testing:

This longer one does not work as supposed

[Route 2 A Barntrup shortend 2.zip](https://github.com/osmandapp/OsmAnd/files/9175555/Route.2.A.Barntrup.shortend.2.zip)


This shorter one(only a few kilometer) does work as expected

[Route 2 A Barntrup shortend 3.zip](https://github.com/osmandapp/OsmAnd/files/9175557/Route.2.A.Barntrup.shortend.3.zip)


So to resume, the mistake is still the same, as explained at the beginning of this thread!!!!
To repeat the navigation setting, follow the track from the nearest point without recalculation.

The only difference to Your installation is the osmand version 4.3 ,which is not available for customer right now.

**Comment by Zirochkabila** [comment]
> Hallo 

Yes you are right

**Comment by Zirochkabila** [comment]
> 4.2.7

Your case was checked on the application version 4.2.7 - it could not be reproduced. As you can see in the screenshot, route re-routing is performed even when navigating on a longer GPX.

<table style="width:100%">
  <tr>
<th><img src="https://user-images.githubusercontent.com/104760013/180959727-493c2d90-77a0-439a-9599-445609916479.jpeg"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by C6competition** [comment]
High

I know the joining is working, if you select RECALCULATION !
The mistake is only taken place in NON RECALCULATION _and_ the track is long enough _and_ the starting position is about half of the way from start to finish!
Your example above I could reproduce easily as well under RECALCULATION & NON RECALCULATION in course of your starting position!

Please let me show you again the problem:

All under NON RECALCULATION conditiones!!!!!!

Please use this track: 
[Route 2 A Barntrup shortend 2.zip](https://github.com/osmandapp/OsmAnd/files/9188693/Route.2.A.Barntrup.shortend.2.zip)

Use start point Welsede / L431 as on screenshot, this is the result

![long_welsede_nonrecalc](https://user-images.githubusercontent.com/99248549/180978988-56117536-bcf9-4bf0-983c-d399fbbf1759.jpg)

This is the expected result with 208 km to go to the finish



Use start point Rothenuffeln / B65 as on screenshot, this is the result

![long_Rothenuffeln_norecalc](https://user-images.githubusercontent.com/99248549/180979354-eb319bf7-b199-4fe7-9759-123eda2c2cc7.jpg)

Here You see the difference, with 257km to go, 
while the whole track is only about 210 km long!
And there is no routing to the track done

Even if I put the start point Rothenuffeln / B65 directly on the track, this is the result

![long_Rothenuffeln_ontrack_norecalc](https://user-images.githubusercontent.com/99248549/180983145-840f4dda-b3e0-4d2c-b999-b12a1a967104.jpg)

You get the same fault, with only a little bit different distance to go to the finish 258 km

**Comment by C6competition** [comment]
High

I think I found the course for the problem:

Again preset is NO RECALCULATION and POINT OF THE TRACK TO NAVIGATE -> NEAREST POINT

For this demonstration I use again this track

[Route 2 A Barntrup shortend 2.zip](https://github.com/osmandapp/OsmAnd/files/9196925/Route.2.A.Barntrup.shortend.2.zip)

and the start point Rothenuffeln / B65


![long_Rothenuffeln_norecalc 1](https://user-images.githubusercontent.com/99248549/181212408-a3379ee5-5138-458e-ba97-e5d67eb7c94b.jpg)

Now I select navigation to get this panel:

![long_Rothenuffeln_norecalc 2](https://user-images.githubusercontent.com/99248549/181212621-4d84216b-d468-4be9-9cb1-0476a9dcb46d.jpg)

On left top top you see 
From: My Position !!!!!
To: Set Destination

Now select the displayed track (Route 2 A Barntrup shortend 2) on the bottom, as track to follow and You get this panel:

![long_Rothenuffeln_norecalc 3](https://user-images.githubusercontent.com/99248549/181213939-000bc4e6-5ee5-4061-af11-9b874165d6f2.jpg)

Watch, as soon as You select Destination the start position gets changed as well!!!!

From: L429 (Start position of the track you want to follow)
To: Route 2 A Barntrup shortend 2


![long_Rothenuffeln_norecalc 4](https://user-images.githubusercontent.com/99248549/181214995-8076e2f7-aba1-4d02-a056-9d2c41403110.jpg)

Of course now the whole track is marked on the screen!

Clicking on the field to enter the start position (From:) gives You this panel, which still tells you, 
the start point is MY POSITION !!!!!


![long_Rothenuffeln_norecalc 5](https://user-images.githubusercontent.com/99248549/181216039-5f5007d2-6fe7-4dc9-a2a1-4adace43af2a.jpg)

If You agree You get this Panel

![long_Rothenuffeln_norecalc 6](https://user-images.githubusercontent.com/99248549/181216342-e7bb64eb-a34e-4ea3-9985-61376e6065af.jpg)

Now everything is, as it should be, Your Start position is Your current position
and Your aim is the end of the selected track

![long_Rothenuffeln_norecalc 7](https://user-images.githubusercontent.com/99248549/181216716-08ff1102-74f1-46d3-ba3a-56b5aed3eab2.jpg)

The bug is, when You select the track to follow, the start point is set to the start point of the track,
instead to the actual positon, as it is selected under POINT OF THE TRACK TO NAVIGATE -> NEAREST POINT
For some reasons this doesn't happen, if the actual position is close enough to the start point of the track to follow ... etc.


To work around this "feature" :

After selecting the track to follow, You have to select Your starting point again !!!!!!!!


Do You agree ?

**Comment by Zirochkabila** [comment]
Thank you for the detailed description - I was able to reproduce all of the above.
You are right, except that when you select GPX your start position is automatically determined by track start position, and in screenshot 5 the action is displayed, not the state - i.e. what you need to do is to have a start position displayed at your actual location, not at the start of GPX (actually this is what happens when you click on My position - the track is re-calculated, as in screenshot 6)

**Comment by C6competition** [comment]
Thank You @Zirochkabila

Yes, Your right and I decided for myself, always to define the start point after selecting the track to follow!

Before I selected from top to bottom: first start point (preset to act position, which was fine) then track to follow.
The strange thing is, the starting point is changed after selecting the track to follow, depending on the actual position you are!
(see example on my  #https://github.com/osmandapp/OsmAnd/issues/14796#issuecomment-1195299201)

This is, what I didn't notice at the beginning!
